pkg/channel/consolidated/reconciler/controller/kafkachannel.go

```diff
@@ -261,29 +262,31 @@ func (r *Reconciler) setupSubscriptionStatusWatcher(ctx context.Context, ch *v1b
 				ObservedGeneration: s.Generation,
 				Ready:              corev1.ConditionTrue,
 			})
+		} else {
+			reconciled = false
```

Comment:
realistically, if we know that we haven't properly reconciled here, why are we even continuing down the rest of the function?